src/Hamiltonians/Molecular.jl

```diff
@@ -0,0 +1,613 @@
+            return false
+        else
+            return true
+        end
+    elseif s.n_excited == (0, 2)
+        if ((1 <= s.from[1] <= length(iter.modes.occupied[2]))
+            &&
+            (1 <= s.from[2] <= length(iter.modes.occupied[2]))
+            &&
+            (1 <= s.to[1] <= length(iter.modes.unoccupied[2]))
+            &&
+            (1 <= s.to[2] <= length(iter.modes.unoccupied[2])))
+            return false
+        else
+            return true
+        end
+    elseif s.n_excited == (2, 0)
+        if ((1 <= s.from[1] <= length(iter.modes.occupied[1]))
+            &&
+            (1 <= s.from[2] <= length(iter.modes.occupied[1]))
+            &&
+            (1 <= s.to[1] <= length(iter.modes.unoccupied[1]))
+            &&
+            (1 <= s.to[2] <= length(iter.modes.unoccupied[1])))
+            return false
+        else
+            return true
+        end
+    elseif s.n_excited == (1, 1)
+        if ((1 <= s.from[1] <= length(iter.modes.occupied[1]))
+            &&
+            (1 <= s.from[2] <= length(iter.modes.occupied[2]))
+            &&
+            (1 <= s.to[1] <= length(iter.modes.unoccupied[1]))
+            &&
+            (1 <= s.to[2] <= length(iter.modes.unoccupied[2])))
+            return false
+        else
+            return true
+        end
+    else
+        return true
+    end
+end
+
+function MolecularHamiltonianOffDiagonalsIteratorState(
+    n_excited::Tuple{Int,Int}, ii::Int, ij::Int
+)
+    return MolecularHamiltonianOffDiagonalsIteratorState(
+        n_excited, (ii, 0), (ij, 0)
+    )
+end
+
+function MolecularHamiltonianOffDiagonalsIteratorState(
+    n_excited::Tuple{Int,Int},
+    ii::Int, ij::Int, ik::Int, il::Int
+)
+    return MolecularHamiltonianOffDiagonalsIteratorState(
+        n_excited, (ii, ij), (ik, il)
+    )
+end
+
+function MolecularHamiltonianOffDiagonalsIteratorState(na::Int, nb::Int)
+    if (na, nb) == (0, 1) || (na, nb) == (1, 0)
+        return MolecularHamiltonianOffDiagonalsIteratorState((na, nb), (1, 0), (1, 0))
+    elseif (na, nb) == (0, 2) || (na, nb) == (2, 0)
+        return MolecularHamiltonianOffDiagonalsIteratorState((na, nb), (1, 2), (1, 2))
+    elseif (na, nb) == (1, 1)
+        return MolecularHamiltonianOffDiagonalsIteratorState((na, nb), (1, 1), (1, 1))
+    else
+        return MolecularHamiltonianOffDiagonalsIteratorState((na, nb), (0, 0), (0, 0))
+    end
+end
+
+function Base.iterate(iter::MolecularHamiltonianOffDiagonalsIterator)
+    initial = MolecularHamiltonianOffDiagonalsIteratorState(0, 1)
+    return iterate(iter, initial)
+end
+
+function Base.iterate(
+    iter::MolecularHamiltonianOffDiagonalsIterator,
+    state::MolecularHamiltonianOffDiagonalsIteratorState
+)
+    na, nb = state.n_excited
+    if (na, nb) == (0, 1)
+        if is_void_state(state) || is_invalid_state(iter, state)
+            return iterate(
+                iter, MolecularHamiltonianOffDiagonalsIteratorState(1, 0)
```

Comment:
```suggestion
                iter, MolecularHamiltonianOffDiagonalsIteratorState((1, 0), (1, 0), (1, 0))
```
See comment above!

Please apply to all the other cases as well!